File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:

        vowels={'a','e','i','o','u'}

        current_count=0
        max_count=0
        left=0

        for i in range(k):
            if s[i] in vowels:
                current_count+=1
        max_count=current_count

        for r in range(k,len(s)):
            if s[r] in vowels:
                current_count+=1
            if r-left+1 >k:
                if s[left] in vowels:
                    current_count-=1
                left+=1
            max_count=max(max_count,current_count)
        return max_count





























        vowels={'a','e','i','o','u'}

        max_count,current_count=0,0

        for i in range(k):
            if s[i] in vowels:
                current_count+=1
        max_count=current_count

        for r in range(k,len(s)):
            if s[r] in vowels:
                current_count+=1
            if s[r-k] in vowels:
                current_count-=1
            max_count=max(max_count,current_count)
        return max_count

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {

        Set <Character> vowels=new HashSet<>();
        vowels.add('a');
        vowels.add('e');
        vowels.add('i');
        vowels.add('o');
        vowels.add('u');


        int max_count=0;
        int left=0;
        int current_count=0;

        for(int i=0;i<k;++i){
            if(vowels.contains(s.charAt(i))){
                current_count++;
            }
        }
        max_count=current_count;
        for(int i=k;i<s.length();++i){
            if(vowels.contains(s.charAt(i))){
                current_count++;
            }
            if(vowels.contains(s.charAt(i-k))){
                current_count--;
            }
            max_count=Math.max(max_count,current_count);
        }
        return max_count;
    }
}